### PR TITLE
Use upstream RBAC in our roles

### PR DESCRIPTION
**Pull request by 0xmichalis**
/cc enj

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@enj

File: cluster/ci/config/prow/prowjob_access.yaml [hunk 1]
```diff
@@ -2,16 +2,17 @@ kind: List
 apiVersion: v1
 items:
 - kind: ClusterRoleBinding
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (enj) [inline cluster/ci/config/prow/prowjob_access.yaml]: This files diff should be: https://www.diffchecker.com/s3z7AEMx

File: cluster/ci/config/roles.yaml [hunk 2]
```diff
@@ -39,40 +39,44 @@ items:
...
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: ci-admins-sudoers
   subjects:
   - kind: Group
     name: ci-admins
   roleRef:
+    apiGroup: rbac.authorization.k8s.io
     name: sudoer
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: `kind: ClusterRole`

File: cluster/ci/config/roles.yaml [hunk 3]
```diff
@@ -39,40 +39,44 @@ items:
   - dustymabe
   - jlebon
 - kind: ClusterRoleBinding
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: ci-admins-sudoers
   subjects:
   - kind: Group
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: `apiGroup: rbac.authorization.k8s.io`

File: cluster/ci/config/roles.yaml [hunk 4]
```diff
@@ -39,40 +39,44 @@ items:
...
   roleRef:
+    apiGroup: rbac.authorization.k8s.io
     name: sudoer
 - kind: ClusterRoleBinding
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: ci-admins-to-cluster-reader
   subjects:
   - kind: Group
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: `apiGroup: rbac.authorization.k8s.io`

File: cluster/ci/config/roles.yaml [hunk 5]
```diff
@@ -39,40 +39,44 @@ items:
...
     name: ci-admins
   roleRef:
+    apiGroup: rbac.authorization.k8s.io
     name: cluster-reader
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: `kind: ClusterRole`

File: cluster/ci/config/roles.yaml [hunk 6]
```diff
@@ -39,40 +39,44 @@ items:
...
 - kind: RoleBinding
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
   metadata:
     name: authenticated-users-to-view
     namespace: ci
   subjects:
   - kind: SystemGroup
```
Comment (enj) [inline cluster/ci/config/roles.yaml]:
```
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:authenticated
```

File: cluster/ci/config/roles.yaml [hunk 7]
```diff
@@ -39,40 +39,44 @@ items:
...
     name: system:authenticated
   roleRef:
+    apiGroup: rbac.authorization.k8s.io
     name: view
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: `kind: ClusterRole`

File: cluster/ci/config/roles.yaml [hunk 8]
```diff
@@ -39,40 +39,44 @@ items:
...
 - kind: RoleBinding
-  apiVersion: v1
+  apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (enj) [inline cluster/ci/config/roles.yaml]: Same last 2 changes since this is the same role binding in different namespaces.

**Comment by kargakis** [comment]
@enj thanks for the review, comments addressed.

Should I also update the Group apiVersion from `v1` to `rbac.authorization.k8s.io/v1beta1` below?  https://github.com/openshift/release/blob/934aae73f6f11e9137ca234ac67b98cae5d7e28f/cluster/ci/config/roles.yaml#L22-L40

**Comment by enj** [comment]
> Should I also update the Group apiVersion from `v1` to `rbac.authorization.k8s.io/v1beta1` below?

No, those are API types that only exist in OpenShift (and get insert into the user info and are treated that same as the RBAC groups).

The changes LGTM, but you should test them against a real sever to make sure they work as you expect.

**Comment by kargakis** [comment]
> No, those are API types that only exist in OpenShift (and get insert into the user info and are treated that same as the RBAC groups).

Then why do I need to add it in the clusterrolebinding?
```yaml
   - kind: Group
+    apiGroup: rbac.authorization.k8s.io
     name: ci-admins
   roleRef:
+    kind: ClusterRole
+    apiGroup: rbac.authorization.k8s.io
     name: prowjob-access
```

**Comment by enj** [comment]
> Then why do I need to add it in the clusterrolebinding?

Because that refers to an RBAC virtual group, not the Group API object from OpenShift.

**Comment by stevekuznetsov** [comment]
> Because that refers to an RBAC virtual group, not the Group API object from OpenShift.

But of course

**Comment by enj** [comment]
> But of course

I know, I know.  Just close your eyes and press the green button.

**Comment by kargakis** [comment]
@deads2k :sad face:

**Comment by deads2k** [comment]
> @deads2k :sad face:

Gives us a hook to build a different resolver if we ever needed one, but we ended up collapsing to match.  I don't miss system:groups.

**Pull request merged**